indexer/feature_altitude.hpp

```diff
@@ -0,0 +1,120 @@
+#pragma once
+#include "coding/reader.hpp"
+#include "coding/varint.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "base/assert.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/limits.hpp"
+#include "std/vector.hpp"
+
+namespace feature
+{
+using TAltitude = int16_t;
+using TAltitudes = vector<feature::TAltitude>;
+
+TAltitude constexpr kInvalidAltitude = numeric_limits<TAltitude>::min();
+
+struct AltitudeHeader
+{
+  using TAltitudeSectionVersion = uint16_t;
+
+  AltitudeHeader() { Reset(); }
+  template <class TSink>
+  void Serialize(TSink & sink) const
+  {
+    WriteToSink(sink, m_version);
+    WriteToSink(sink, m_minAltitude);
+    WriteToSink(sink, m_featureTableOffset);
+    WriteToSink(sink, m_altitudesOffset);
+    WriteToSink(sink, m_endOffset);
+  }
+
+  template <class TSource>
+  void Deserialize(TSource & src)
+  {
+    m_version = ReadPrimitiveFromSource<TAltitudeSectionVersion>(src);
+    m_minAltitude = ReadPrimitiveFromSource<TAltitude>(src);
+    m_featureTableOffset = ReadPrimitiveFromSource<uint32_t>(src);
+    m_altitudesOffset = ReadPrimitiveFromSource<uint32_t>(src);
+    m_endOffset = ReadPrimitiveFromSource<uint32_t>(src);
+  }
+
+  // Methods below return sizes of parts of altitude section in bytes.
+  size_t GetAltitudeAvailabilitySize() const
+  {
+    return m_featureTableOffset - sizeof(AltitudeHeader);
+  }
+  size_t GetFeatureTableSize() const { return m_altitudesOffset - m_featureTableOffset; }
+  size_t GetAltitudeInfo() const { return m_endOffset - m_altitudesOffset; }
+  void Reset()
+  {
+    m_version = 0;
+    m_minAltitude = kInvalidAltitude;
+    m_featureTableOffset = 0;
+    m_altitudesOffset = 0;
+    m_endOffset = 0;
+  }
+
+  TAltitudeSectionVersion m_version;
+  TAltitude m_minAltitude;
+  uint32_t m_featureTableOffset;
+  uint32_t m_altitudesOffset;
+  uint32_t m_endOffset;
+};
+
+static_assert(sizeof(AltitudeHeader) == 16, "Wrong header size of altitude section.");
+
+class Altitudes
+{
+public:
+  Altitudes() = default;
+  explicit Altitudes(TAltitudes const & altitudes) : m_altitudes(altitudes) {}
```

Comment:
Blank lines, please. Seems that clang format removed these blank lines, as far as I know they were here..